apps/web/src/components/Basenames/RegistrationSearchInput/index.tsx

```diff
@@ -223,9 +226,30 @@ export default function RegistrationSearchInput({
     [setSearchInputHovered],
   );
 
+  const handleScroll = () => {
+    const position = window.scrollY;
+    setScrollPosition(position);
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll, { passive: true });
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (scrollPosition > 250) {
+      setHeaderBackground(false);
+    } else {
+      setHeaderBackground(true);
+    }
+  }, [scrollPosition]);
```

Comment:
I think you could simplify this code by doing the following:

```suggestion
useEffect(() => {
  const handleScroll = () => {
    setHeaderBackground(window.scrollY <= 250);
  };

  window.addEventListener('scroll', handleScroll, { passive: true });

  return () => {
    window.removeEventListener('scroll', handleScroll);
  };
}, []);
```

You might also consider some performance optimization here if you notice any jank. Running this callback on every scroll event could get kinda jank.  Calling a setState on every scroll event is a little expensive.

Reply:
Consider looking into debouncing/throttling the handler.